--- a/T80.vhd
+++ b/T80.vhd
@@ -267,7 +267,7 @@
 			 else IntE_FF2 & IntE_FF1 & IStatus & DOR(127 downto 112) & DOR(47 downto 0) & DOR(63 downto 48) & DOR(111 downto 64) &
 						std_logic_vector(PC) & std_logic_vector(SP) & std_logic_vector(R) & I & Fp & Ap & F & ACC;
 
-	mcode : T80_MCode
+	mcode : entity work.T80_MCode(rtl)
 		generic map(
 			Mode   => Mode,
 			Flag_C => Flag_C,
@@ -342,7 +342,7 @@
 			No_PC       => No_PC,
 			XYbit_undoc => XYbit_undoc);
 
-	alu : T80_ALU
+	alu : entity work.T80_ALU(rtl)
 		generic map(
 			Mode   => Mode,
 			Flag_C => Flag_C,
@@ -682,7 +682,7 @@
 						F(Flag_N) <= DI_Reg(7);
 						F(Flag_C) <= ioq(8);
 						F(Flag_H) <= ioq(8);
-						ioq := (ioq and x"7") xor ('0'&BusA);
+						ioq := (ioq and "000000111") xor ('0'&BusA);
 						F(Flag_P) <= not (ioq(0) xor ioq(1) xor ioq(2) xor ioq(3) xor ioq(4) xor ioq(5) xor ioq(6) xor ioq(7));
 					end if;
 
@@ -1076,7 +1076,7 @@
 		end if;
 	end process;
 
-	Regs : T80_Reg
+	Regs : entity work.T80_Reg(rtl)
 		port map(
 			Clk => CLK_n,
 			CEN => ClkEn,
--- a/T80s.vhd
+++ b/T80s.vhd
@@ -112,7 +112,7 @@
 
 begin
 
-	u0 : T80
+	u0 : entity work.T80(rtl)
 	generic map(
 		Mode => Mode,
 		IOWait => IOWait)
